src/twisted/internet/_resolver.py

```diff
@@ -169,6 +170,10 @@ def resolveHostName(self, resolutionReceiver, hostName, portNumber=0,
 
         @return: see interface
         """
+        # If it's Python 2, encode this to IDNA.
+        if not _PY3 and isinstance(hostName, unicode):
+            hostName = hostName.encode('idna')
```

Comment:
1. This should really be using `_idnaBytes` from our own idna wrapper so that (as advertised) we will use the `idna` PyPI package when it is available.
2. I'd really like to remove the conditional and deal with IDNA regardless of Python version; either it's our responsibility to convert non-ASCII into IDNA or not, this is separate from what Python type the ASCII is represented by.
3. As the interface is specified, GAIResolver really ought to be doing this as well.  It's not a regression that it isn't, but it's still a defect, and having these behave inconsistently isn't good.